app/src/main/kotlin/compiler/parser/grammar/Symbol.kt

```diff
@@ -5,13 +5,13 @@ import compiler.lexer.lexer_grammar.TokenType
 sealed class Symbol : Comparable<Symbol> {
     data class Terminal(val tokenType: TokenType) : Symbol() {
         override fun compareTo(other: Symbol): Int {
-            TODO("Not yet implemented")
+            return if (other is Terminal) tokenType.compareTo(other.tokenType) else -1
         }
     }
 
     data class NonTerminal(val nonTerminal: NonTerminalType) : Symbol() {
         override fun compareTo(other: Symbol): Int {
-            TODO("Not yet implemented")
+            return if (other is NonTerminal) nonTerminal.compareTo(other.nonTerminal) else 1
         }
```

Comment:
I feel like it could be useful to have a more concise toString function that just returns the string for tokenType/nonTerminal, maybe with n/t as opposed to the default from data class.

Reply:
I didn't suffer much from the default toString from data class during debugging, but custom one can be added too.